executor/adapter.go

```diff
@@ -269,13 +271,20 @@ func (a *ExecStmt) logSlowQuery() {
 	costTime := time.Since(a.startTime)
 	sql := a.Text
 	if len(sql) > cfg.Log.QueryLogMaxLen {
-		sql = sql[:cfg.Log.QueryLogMaxLen] + fmt.Sprintf("(len:%d)", len(sql))
+		// So the truncate won't make the last character an illegal utf8.
+		length := cfg.Log.QueryLogMaxLen
+		for ; length > 0; length-- {
+			if utf8.RuneStart(sql[length]) || sql[length] <= unicode.MaxASCII {
```

Comment:
`RuneStart` is enough, every ascii character is RuneStart.
```go
for !utf8.RuneStart(sql[length]) {
    length--
}
```
It's impossible every character in the sql string is not rune start.